### PR TITLE
feat(azure-iot-device): 🎸 Implement PnP telemetry

### PR DESCRIPTION
**Pull request by vishnureddy17**

### CODE REVIEW

File: device/core/test/_internal_client_test.js [hunk 1]
```diff
@@ -780,6 +780,82 @@ var ModuleClient = require('../dist/module_client').ModuleClient;
...
+          assert.strictEqual(message.contentType, 'application/json');
+          assert.strictEqual(message.data, JSON.stringify(payload));
+          assert.strictEqual(message.properties.getValue('$.sub'), component);
+          assert.strictEqual(callback, telemetryCallback);
+        }
+        client.sendTelemetry(payload, component, telemetryCallback);
+      });
+    });
   });
 });
```
Comment (anthonyvercolano) [inline device/core/test/_internal_client_test.js:861]: Could you add a test that you can actually await on the sendTelemetry? - would entail you creating fake sendEvent that returns a valid result.

**Review by anthonyvercolano — APPROVED** [review]

File: device/core/test/_internal_client_test.js (hunk 1 above)
Comment (vishnureddy17) [reply device/core/test/_internal_client_test.js:861]: On it

Comment (vishnureddy17) [reply device/core/test/_internal_client_test.js:899]: Done

File: device/core/test/_internal_client_test.js [hunk 2]
```diff
@@ -780,6 +780,120 @@ var ModuleClient = require('../dist/module_client').ModuleClient;
...
+            assert.isUndefined(callback);
+          } catch (err) {
+            return Promise.reject(err);
+          }
+          return Promise.resolve(new results.MessageEnqueued());
+        }
+        client.sendTelemetry(payload).then(
+          function () {
+            testCallback();
+          },
```
Comment (anthonyvercolano) [inline device/core/test/_internal_client_test.js:821]: Isn't the normal pattern to have a .catch in the error case?

Comment (vishnureddy17) [reply device/core/test/_internal_client_test.js:821]: .catch is just syntactic sugar for the two argument .then

**Pull request merged**